File: plugin/src/main/java/ru/arc/ai/tools/Tools.java
package ru.arc.ai.tools;

import java.util.Collection;
import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class Tools {

    static {
        tools = new ConcurrentHashMap<>();
        addTool(DefaultTools.LeaveForTime.class);
        addTool(DefaultTools.GetBalTop.class);
        addTool(DefaultTools.GetPlayerInfo.class);
    }

    private static final Map<String, Class<? extends Tool>> tools;

    public static void addTool(Class<? extends Tool> toolClass) {
        tools.put(toolClass.getSimpleName().toLowerCase(), toolClass);
    }

    public static Class<? extends Tool> getTool(String name) {
        return tools.get(name.toLowerCase());
    }

    public static Collection<Class<? extends Tool>> getAllTools() {
        return tools.values();
    }
}

File: src/main/kotlin/ru/arc/ai/tools/Tools.kt
package ru.arc.ai.tools

import java.util.concurrent.ConcurrentHashMap

object Tools {
    private val tools: MutableMap<String, Class<out Tool>> = ConcurrentHashMap()

    init {
        addTool(DefaultTools.RememberFact::class.java)
        addTool(DefaultTools.ForgetFact::class.java)
        addTool(DefaultTools.LeaveForTime::class.java)
        addTool(DefaultTools.GetBalTop::class.java)
        addTool(DefaultTools.GetPlayerInfo::class.java)
        addTool(DefaultTools.GetInventory::class.java)
    }

    @JvmStatic
    fun addTool(toolClass: Class<out Tool>) {
        tools[toolClass.simpleName.lowercase()] = toolClass
    }

    @JvmStatic
    fun getTool(name: String): Class<out Tool>? = tools[name.lowercase()]

    @JvmStatic
    fun getAllTools(): Collection<Class<out Tool>> = tools.values
}
